kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js

```diff
@@ -145,23 +151,27 @@ grCollectionsPanel.directive('grNode', ['$parse', '$compile', function($parse, $
         controller: 'GrNodeCtrl',
         controllerAs: 'ctrl',
         bindToController: true,
-        link: function(scope, element) {
-            // We compile the template on the fly here as angular doesn't deal
-            // well with recursive templates.
-            $compile(`<gr-nodes
-                ng:if="ctrl.showChildren && ctrl.node.data.children.length > 0"
-                gr:selected-images="ctrl.selectedImages$"
-                gr:selected-collections="ctrl.selectedCollections"
-                gr:editing="ctrl.editing"
-                gr:nodes="ctrl.node.data.children"></gr-nodes>
-            `)(scope, cloned => {
-                element.find('.node__children').append(cloned);
-            });
+        compile: function(element, attrs, transclude) {
+            // Memoize the `$compile` result for performance reasons
+            // (compile invoked once per reference, link invoked once
+            // per use)
+            let compiledTemplate;
+            return function link(scope, element) {
+                if (! compiledTemplate) {
+                    // We compile the template on the fly here as angular doesn't deal
+                    // well with recursive templates.
+                    compiledTemplate = $compile(templateString);
+                }
+
+                compiledTemplate(scope, cloned => {
+                    element.find('.node__children').append(cloned);
```

Comment:
I'm not entirely sure where `node__children` is coming from or found, actually. I haven't changed that code.

@jamesgorrie ?